Docker logins don't work on forks

## Describe the bug

Due to how Travis exposes secrets to builds and these not being available on forks the `docker login` command perform before executing the integration tests fails due to the following error:

```
[INFO] Logging into Docker Hub Enabled=true
echo  | docker login -u  --password-stdin
Error: Cannot perform an interactive login from a non TTY device
make: *** [dockerhub-login] Error 1
```

This works if the build is triggered by the repo maintainer as Travis CI trusts the source, however forks don't have the secrets available to them and this is causing all the PRs opened from forks to fail.

The reason why the `docker login` was added in the first place was to avoid getting [quota rate limits](https://www.docker.com/increase-rate-limits/) from docker when pulling images and this does work for PRs opened by the repo maintainers but not for forks. Unfortunately, Travis CI does not provide a solution for the aforementioned issue and the preferable solution until a solution is available is to not perform the `docker login`. This means that PRs from forks will rely on whether the rate limit has already been reached or not but that is preferable over always failing.

Screenshot from Travis CI stating that secrets are not avail in forks:

<img width="875" alt="Screen Shot 2022-04-09 at 4 28 27 PM" src="https://user-images.githubusercontent.com/1503106/162595200-acb53b22-80f2-4986-8e2f-c5166d15cd88.png">
Source: https://docs.travis-ci.com/user/environment-variables/

Increase limits from docker: https://www.docker.com/increase-rate-limits/

## To Reproduce
A clear and concise description on how does it work now and what's the current behaviour.

Steps to reproduce the behavior:
1. Fork the repo
2. Create a PR
3. The build will fail due to the following error:

````
[INFO] Logging into Docker Hub Enabled=true
echo  | docker login -u  --password-stdin
Error: Cannot perform an interactive login from a non TTY device
make: *** [dockerhub-login] Error 1
The command "make test-all" exited with 2.
````

Example build failing from a fork: https://app.travis-ci.com/github/dikhan/terraform-provider-openapi/builds/249184166

## Expected behaviour

Builds should work regardless of whether the build is trigger from maintainers or external contributors. Due to Travis CI limitations though this is not possible at the moment and might need to consider other CI services.

## Additional context
Add any other context about the problem here.

### Checklist (for admin only)

**Don't forget to go through the checklist to make sure the issue is created properly**:

- [ ] I have added a corresponding label (bug) to the issue (right side menu)
- [ ] I have added this issue to the 'API Terraform Provider' GitHub project (right side menu)

--- a/.travis.yml
+++ b/.travis.yml
@@ -10,7 +10,7 @@ services:
   - docker
 
 env:
-  - TF_VERSION=0.13.6 PERFORM_DOCKER_LOGIN=true GO111MODULE=on
+  - TF_VERSION=0.13.6 PERFORM_DOCKER_LOGIN=$TRAVIS_SECURE_ENV_VARS GO111MODULE=on
 
 install:
   - go install github.com/goreleaser/goreleaser@v1.6.3

